velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/dwio/common/Reader.h"
+#include "velox/dwio/parquet/reader/ParquetReader.h"
+#include "velox/dwio/parquet/tests/ParquetReaderTestBase.h"
+#include "velox/dwio/parquet/writer/Writer.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::common;
+using namespace facebook::velox::test;
+
+namespace facebook::velox::parquet {
+class ParquetWriterTest : public dwio::parquet::ParquetReaderTestBase,
+                          public VectorTestBase {
+ public:
+  ParquetWriterTest()
+      : rootPool_(
+            memory::defaultMemoryManager().addRootPool("ParquetWriterTest")),
+        leafPool_(rootPool_->addLeafChild("ParquetWriterTest")){};
+
+  std::unique_ptr<parquet::Writer> createWriter(
+      parquet::WriterOptions& writerOpts) {
+    auto sink = std::make_unique<dwio::common::MemorySink>(
```

Comment:
Its simpler to write to a local file and read from a local file.
Using memory is odd since you can only create the reader after writing to memory.

Reply:
So should I write to a local file instead?